src/trunk.jl

```diff
@@ -135,8 +151,10 @@ function solve!(
     error("trunk should only be called for unconstrained problems. Try tron instead")
   end
 
+  wks = Dict{Symbol,Any}(:x => x)
```

Comment:
Would it make sense if `wks` is a `GenericExecutionStats`?

Reply:
We would need to add more things to `GenericExecutionStats`, also maybe we want to expose more things that don't make sense for either the solver or the stats.
I don't discard the option though, so I think we can discuss this.

Reply:
The additional entries could go to the `solver_specific`.
and the `status` option might be sufficient instead of the various booleans.

`GenericExecutionStats` would also give access to the counters.

It simplifies the code if we plan to pre-allocate the stats.

Reply:
That makes sense. We'll need some refactoring to update the `status` earlier and to handle the `:user` output, but it seems feasible. Let me try to change it a little.